sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuRoundRobinPartitioning.scala

```diff
@@ -83,12 +98,16 @@ case class GpuRoundRobinPartitioning(numPartitions: Int)
     }
   }
 
-  private def getStartPartition : Int = {
+  private def getStartPartition: Int = {
     // Follow Spark's way of getting the randomized position to pass to the round robin,
     // which starts at randomNumber + 1 and we do the same here to maintain compatibility.
     // See ShuffleExchangeExec.getPartitionKeyExtractor for reference.
-    val position = new Random(TaskContext.get().partitionId()).nextInt(numPartitions) + 1
-    val partId = position.hashCode % numPartitions
-    partId
+    val random = if (null != TaskContext.get()) {
+      new Random(TaskContext.get().partitionId())
+    } else {
+      // For unit test purpose where task context does not exist
+      new Random
```

Comment:
nit: Should we have a hard coded seed here instead? just so that we get deterministic results for unit tests.

Reply:
Yes, it could be helpful when we need to validate result. We can touch this part when we have such needs.